lib/mlir/Dialect/Clift/IR/CliftOps.cpp

```diff
@@ -806,3 +806,227 @@ mlir::LogicalResult YieldOp::verify() {
+                              " size.";
+  } break;
+  case CastKind::Decay: {
+    auto PtrT = mlir::dyn_cast<PointerType>(ResT);
+    if (not PtrT)
+      return emitOpError() << getOperationName()
+                           << " result must have pointer type.";
+
+    if (auto ArrayT = mlir::dyn_cast<ArrayType>(ArgT)) {
+      if (PtrT.getPointeeType() != ArrayT.getElementType())
+        return emitOpError() << getOperationName()
+                             << " the pointee type of the result type must be"
+                                " equal to the element type of the argument"
+                                " type.";
+    } else if (auto DefinedT = mlir::dyn_cast<DefinedType>(ArgT)) {
+      auto const
+        FunctionT = mlir::dyn_cast<FunctionTypeAttr>(DefinedT.getElementType());
+
+      if (not FunctionT)
+        return emitOpError() << getOperationName()
+                             << " argument must have array or function type.";
+
+      if (PtrT.getPointeeType() != DefinedT)
+        return emitOpError() << getOperationName()
+                             << " the pointee type of the result type must be"
+                                " equal to the argument type.";
+    } else {
+      return emitOpError() << getOperationName()
+                           << " argument must have array or function type.";
+    }
+  } break;
+
+  default:
+    revng_abort("Invalid CastKind value");
+  }
+
+  return mlir::success();
+}
+
+//===----------------------------- AddressofOp ----------------------------===//
+
+mlir::LogicalResult AddressofOp::verify() {
+  if (not clift::isLvalueExpression(getObject()))
+    return emitOpError() << getOperationName()
+                         << " operand must be an lvalue-expression.";
+
+  return mlir::success();
+}
+
+//===---------------------------- IndirectionOp ---------------------------===//
+
+mlir::LogicalResult IndirectionOp::verify() {
+  if (isVoid(getResult().getType()))
+    return emitOpError() << getOperationName()
+                         << " cannot dereference a pointer to void.";
+
+  return mlir::success();
+}
+
+//===------------------------------ AssignOp ------------------------------===//
+
+mlir::LogicalResult AssignOp::verify() {
+  auto Type = mlir::cast<clift::ValueType>(getLhs().getType());
+  auto [UnderlyingType, HasConst] = decomposeTypedef(Type);
+
+  if (HasConst or UnderlyingType.isConst())
+    return emitOpError() << getOperationName()
+                         << " left operand type must be non-const.";
+
+  if (not clift::isLvalueExpression(getLhs()))
+    return emitOpError() << getOperationName()
+                         << " left operand must be an lvalue-expression.";
+
+  return mlir::success();
+}
+
+//===--------------------------- ObjectAccessOp ---------------------------===//
+
+static ValueType getField(ValueType ClassType, size_t Index) {
+  const auto GetField = [&](llvm::ArrayRef<FieldAttr> Fields) -> ValueType {
```

Comment:
When the scope of the lambda is so limited and its lifetime so well defined, I much prefer the automatic reference capture, because it really doesn't matter. Explicitly listing the captures is just noise that tricks the reader into thinking it actually matters.

Reply:
I always prefer to restrict as much as possible access to external context inside a lambda. And capturing everything by reference doesn't restrict anything.

But I agree that this lambda is so small that it makes little difference.

Like I said, not very important. Do what you prefer in this case, and close this thread when you feel like it.